src/main/java/org/shanerx/tradeshop/listeners/ShopTradeListener.java

```diff
@@ -55,189 +60,222 @@ public void onBlockInteract(PlayerInteractEvent e) {
 		Sign s;
 		BlockState chestState;
 
-		if (e.getAction() == Action.RIGHT_CLICK_BLOCK) {
+		if (ShopType.isShop(e.getClickedBlock())) {
+			s = (Sign) e.getClickedBlock().getState();
+		} else {
+			return;
+		}
 
-			if (ShopType.isShop(e.getClickedBlock())) {
-				s = (Sign) e.getClickedBlock().getState();
-			} else {
-				return;
-			}
+		JsonConfiguration json = new JsonConfiguration(s.getChunk());
+		shop = json.loadShop(new ShopLocation(s.getLocation()));
 
-			JsonConfiguration json = new JsonConfiguration(s.getChunk());
-			shop = json.loadShop(new ShopLocation(s.getLocation()));
+		if (shop == null)
```

Comment:
Just because the sign is a shop sign doesnt mean the there is still a shop saved in file, this is to make sure the shop was actually loaded otherwise it throws NPEs